### PR TITLE
Update electron to 7 and electron related packages to new versions

### PR DESCRIPTION
**Pull request by ashchan**
Experiment electron 7 upgrade.

Note: this isn't going to be released with `v0.26.0`.

### CODE REVIEW

File: packages/neuron-wallet/src/controllers/app/index.ts [hunk 1]
```diff
@@ -6,14 +6,20 @@ import env from 'env'
...
 const app = electronApp || (remote && remote.app)
 
 export default class AppController {
   public mainWindow: BrowserWindow | null
+  apiController: ApiController
 
   constructor() {
     this.mainWindow = null
+
+    this.apiController = new ApiController
```
Comment (Keith-CY) [inline packages/neuron-wallet/src/controllers/app/index.ts]:
```suggestion
    this.apiController = new ApiController()
```

File: packages/neuron-ui/src/services/remote/remoteApiWrapper.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  | 'check-for-updates'
+  | 'download-update'
+  | 'quit-and-install-update'
+  // Settings
+  | 'clear-cache'
+
+export const remoteApi = <T = any>(action: Action) => async (params: T): Promise<ControllerResponse> => {
+  const res: SuccessFromController | FailureFromController = await window.ipcRenderer
+    .invoke(action, params)
+    .then(stringifiedRes => (stringifiedRes ? JSON.parse(stringifiedRes) : stringifiedRes))
```
Comment (Keith-CY) [inline packages/neuron-ui/src/services/remote/remoteApiWrapper.ts]:
```suggestion
```
> Send a message to the main process asynchronously via channel and expect an asynchronous result. Arguments will be serialized as JSON internally and hence no functions or prototype chain will be included.

File: packages/neuron-wallet/src/controllers/api.ts [hunk 3]
```diff
@@ -1,333 +1,282 @@
...
-  @MapApiResponse
-  public static async clearCellCache() {
-    return SyncController.clearCache()
+  // Register handler, warp and serialize API response
+  static NODE_DISCONNECTED_CODE = 104
+  private handleChannel(channel: string, listener: (event: IpcMainInvokeEvent, ...args: any[]) => (Promise<void>) | (any)) {
+    ipcMain.handle(channel, async (event, args) => {
+      try {
+        const res = await listener(event, args)
+        return JSON.stringify(res)
```
Comment (Keith-CY) [inline packages/neuron-wallet/src/controllers/api.ts]:
```suggestion
        return res
```

File: packages/neuron-wallet/src/controllers/api.ts [hunk 4]
```diff
@@ -1,333 +1,282 @@
...
+      } catch (err) {
+        if (err.code === 'ECONNREFUSED') {
+          err.code = ApiController.NODE_DISCONNECTED_CODE
+        }
+        const res = {
+          status: err.code || ResponseCode.Fail,
+          message: typeof err.message === 'string' ? { content: err.message } : err.message,
+        }
+        return JSON.stringify(res)
```
Comment (Keith-CY) [inline packages/neuron-wallet/src/controllers/api.ts]:
```suggestion
        return res
```

File: packages/neuron-ui/src/services/remote/remoteApiWrapper.ts (hunk 2 above)
Comment (ashchan) [reply packages/neuron-ui/src/services/remote/remoteApiWrapper.ts]: Nice, let me check and test that.

**Pull request merged**